### ISSUE TITLE
Editing Deployment image names via "Edit Config" results in failure

### ISSUE THREAD

**Issue by nickgerace** [rancher/dashboard#3827]
When editing a deployment's image name, via "Edit Config" errors occur. However, when editing it via "Edit YAML", no errors occur.

Error that appears in UI:
```
Deployment.apps "fleet-agent" is invalid: spec.template.spec.affinity.nodeAffinity.requiredDuringSchedulingIgnoredDuringExecution.nodeSelectorTerms: Required value: must have at least one node selector term
```

Changing this one field appears to create a large diff as well when using "Edit Config". One field is offset too:

<img width="721" alt="Screen Shot 2021-08-12 at 4 00 24 PM" src="https://user-images.githubusercontent.com/39320683/129261701-baca4277-4aa8-4c13-856a-2ebc327d369d.png">

**Comment by nwmac** [comment]
@Sean-McQ Check with @rak-phillip - I think he looked at a similar issue elsewhere (I could be wrong)

**Comment by gaktive** [comment]
There are additional reports about this happening. Internal reference: SURE-3676.

**Comment by gaktive** [comment]
From the internal reference, there's a comment in place tied to `preference` that may be the culprit.

**Comment by Sean-McQ** [comment]
Was unable to reproduce and after a brief conversation with @mantis-toboggan-md  it looks like this was previously resolved in 2.6.3 under https://github.com/rancher/dashboard/pull/4319

**Comment by sgapanovich** [comment]
#### Reproduction ####
Rancher version: 2.6.3
Rancher cluster type: single-node docker install

Downstream cluster type: rke2 ec2 node driver
Downstream K8s version: 1.21.9

1. Workloads -> Deployments -> Create 
2. Name: test
  Container name: test
  Image: nginx
  Node Scheduling: Run pods on node(s) matching scheduling rules
  -Priority: Preferred
  -Key: node-role.kubernetes.io/worker
  -Operator: is set
![test repro](https://user-images.githubusercontent.com/40282141/157101784-1917d6a5-053f-4fc5-910b-4a462132af8b.jpg)

-----------

### Test Environment: ###
Rancher version: v2.6-head 4df2214
Rancher cluster type:  HA
Docker version: 20.10

Downstream cluster type: rke1 ec2 node driver 
Downstream K8s version: 1.23.4

### Testing: ###

Edit Config for:
- deployment with Node Scheduling -> Run pods on nodes(s) matching scheduling rules -> Preferred (affinity)
- deployment with Node Scheduling -> Run pods on nodes(s) matching scheduling rules -> Required (affinity)
- deployment with Node Scheduling -> Run pods on specific node(s) -> Select node (nodeName)

All deployments are updated successfully with pods recreated
